### PR TITLE
Separating this API

### PR DESCRIPTION
**Pull request by alxdarksage**

### CODE REVIEW

File: src/main/java/org/sagebionetworks/bridge/spring/controllers/StudyParticipantController.java [hunk 1]
```diff
@@ -186,6 +191,24 @@ private boolean isUpToDate(DateTime modifiedSince, DateTime modifiedOn) {
...
     }
     
+    @PostMapping("/v5/studies/{studyid}/participants/emailRoster")
+    @ResponseStatus(HttpStatus.ACCEPTED)
+    public StatusMessage getParticipantRoster(@PathVariable String studyId) throws JsonProcessingException {
+        UserSession session = getAuthenticatedSession(STUDY_COORDINATOR);
+
+        if (!RequestContext.get().getOrgSponsoredStudies().contains(studyId)) {
+            throw new UnauthorizedException();   
+        }
```
Comment (alxdarksage) [inline src/main/java/org/sagebionetworks/bridge/spring/controllers/StudyParticipantController.java]: To be consistent, this should be an auth rule (canAccessStudy()).

File: src/main/java/org/sagebionetworks/bridge/spring/controllers/StudyParticipantController.java [hunk 2]
```diff
@@ -186,6 +191,24 @@ private boolean isUpToDate(DateTime modifiedSince, DateTime modifiedOn) {
...
+        StudyParticipant participant = session.getParticipant();
+        if (participant.getEmail() == null || !TRUE.equals(participant.getEmailVerified())) {
+            throw new BadRequestException("A valid email address is required to send the requested participant roster.");
+        }
+        ParticipantRosterRequest request = parseJson(ParticipantRosterRequest.class);
+        participantService.getParticipantRoster(session.getAppId(), session.getId(), request);
```
Comment (alxdarksage) [inline src/main/java/org/sagebionetworks/bridge/spring/controllers/StudyParticipantController.java]: Consolidate email check? So we should pass the callerId in, or that's in the context, and do the check there.

File: src/test/java/org/sagebionetworks/bridge/spring/controllers/StudyParticipantControllerTest.java [hunk 3]
```diff
@@ -221,6 +228,67 @@ public void verifyAnnotations() throws Exception {
         assertDelete(StudyParticipantController.class, "deleteActivityEvent");
```
Comment (alxdarksage) [inline src/test/java/org/sagebionetworks/bridge/spring/controllers/StudyParticipantControllerTest.java:228]: add an assertion of the annotations

File: src/test/java/org/sagebionetworks/bridge/spring/controllers/StudyParticipantControllerTest.java [hunk 4]
```diff
@@ -221,6 +228,67 @@ public void verifyAnnotations() throws Exception {
...
+        RequestContext.set(new RequestContext.Builder()
+                .withOrgSponsoredStudies(ImmutableSet.of(TEST_STUDY_ID)).build());
+        
+        session.setParticipant(new StudyParticipant.Builder()
+                .withId(TEST_USER_ID).withEmail(EMAIL).withEmailVerified(true).build());
+        
+        doReturn(session).when(controller).getAuthenticatedSession(STUDY_COORDINATOR);
+        
+        mockRequestBody(mockRequest, 
+                createJson("{'studyId':'"+TEST_STUDY_ID+"', 'password': '"+PASSWORD+"'}"));
```
Comment (alxdarksage) [inline src/test/java/org/sagebionetworks/bridge/spring/controllers/StudyParticipantControllerTest.java]: This should be overridden by the studyId in the route. that's what we're checking for access.

File: src/test/java/org/sagebionetworks/bridge/spring/controllers/StudyParticipantControllerTest.java [hunk 5]
```diff
@@ -221,6 +228,67 @@ public void verifyAnnotations() throws Exception {
...
+    
+    @Test(expectedExceptions = UnauthorizedException.class)
+    public void getParticipantRoster_notAssociatedToStudy() throws Exception {
+        session.setParticipant(new StudyParticipant.Builder()
+                .withId(TEST_USER_ID).withEmail(EMAIL).withEmailVerified(true).build());
+        
+        doReturn(session).when(controller).getAuthenticatedSession(STUDY_COORDINATOR);
+        
+        mockRequestBody(mockRequest, 
+                createJson("{'studyId':'"+TEST_STUDY_ID+"', 'password': '"+PASSWORD+"'}"));
```
Comment (alxdarksage) [inline src/test/java/org/sagebionetworks/bridge/spring/controllers/StudyParticipantControllerTest.java:262]: Again, you're not using the value in the JSON body.

File: src/main/java/org/sagebionetworks/bridge/spring/controllers/StudyParticipantController.java (hunk 1 above)
Comment (alxdarksage) [reply src/main/java/org/sagebionetworks/bridge/spring/controllers/StudyParticipantController.java]: Changed to auth rule.

File: src/main/java/org/sagebionetworks/bridge/spring/controllers/StudyParticipantController.java (hunk 2 above)
Comment (alxdarksage) [reply src/main/java/org/sagebionetworks/bridge/spring/controllers/StudyParticipantController.java]: Moved to the service and out of the controllers.

File: src/test/java/org/sagebionetworks/bridge/spring/controllers/StudyParticipantControllerTest.java (hunk 4 above)
Comment (alxdarksage) [reply src/test/java/org/sagebionetworks/bridge/spring/controllers/StudyParticipantControllerTest.java]: Changed this for study coordinators (for researchers they can request any study, or no study)

File: src/test/java/org/sagebionetworks/bridge/spring/controllers/StudyParticipantControllerTest.java (hunk 5 above)
Comment (alxdarksage) [reply src/test/java/org/sagebionetworks/bridge/spring/controllers/StudyParticipantControllerTest.java:262]: Changed

File: src/test/java/org/sagebionetworks/bridge/services/ParticipantServiceTest.java [hunk 6]
```diff
@@ -2542,39 +2542,60 @@ public void createCustomActivityEventAccountNotFound() throws Exception {
...
-        participantService.getParticipantRoster(TEST_APP_ID, TEST_USER_ID, request);
+        participantService.requestParticipantRoster(APP, TEST_USER_ID, request);
 
         String requestJson = "{\"service\":\"DownloadParticipantRosterWorker\",\"body\":{\"appId\":\"test-app\"," +
                 "\"userId\":\"userId\",\"password\":\"P@ssword1\",\"studyId\":\"studyId\"}}";
         verify(sqsClient).sendMessage(queueUrl, requestJson);
     }
 
+    @Test(expectedExceptions = BadRequestException.class)
+    public void getParticipantRoster_emailNotVerified() throws JsonProcessingException {
```
Comment (jnelsonjava) [inline src/test/java/org/sagebionetworks/bridge/services/ParticipantServiceTest.java]: This is nit-picky but could rename the tests from "get..." to "request..." to match the method rename.

**Review by jnelsonjava — APPROVED** [review]
Looks good. Only one optional note about renaming.

File: src/test/java/org/sagebionetworks/bridge/services/ParticipantServiceTest.java (hunk 6 above)
Comment (alxdarksage) [reply src/test/java/org/sagebionetworks/bridge/services/ParticipantServiceTest.java]: Updated all these tests with new method name. (not nit-picky!)

**Pull request merged**